devito/finite_differences/tools.py

```diff
@@ -128,14 +128,14 @@ def deriv_function(expr, deriv_order, dims, fd_order, side=centered, **kwargs):
     # Add non-conventional, non-centered first-order FDs
     for d, o in zip(dimensions, orders):
         name = 't' if d.is_Time else d.root.name
-        if function.is_Staggered:
+        if function.is_Staggered or o == 2:
```

Comment:
so if `o == 2` **and** the function isn't staggered, we will enter two `if`s -- this one and the one at line 138 ?

Reply:
yes